ubuntudesign/documentation_builder/operations.py

```diff
@@ -190,38 +190,43 @@ def parse_markdown(parser, template, filepath, metadata):
     return template.render(metadata)
 
 
-def prepare_branches(base_directory, output_base, versions=False):
+def prepare_version_branches(base_directory, output_base):
     """
     If build_version_branches is true, look for a "versions" file in the
     base_directory and then pull each version branch.
     Otherwise, just return the base directory.
     """
 
-    branch_paths = []
+    version_branches = {}
 
-    if not path.isdir(base_directory):
-        raise FileNotFoundError(
-            'Base directory not found: {}'.format(base_directory)
-        )
+    with open(path.join(base_directory, 'versions')) as versions_file:
+        lines = versions_file.read().splitlines()
+        version_branch_names = list(filter(None, lines))
 
-    if versions:
-        with open(path.join(base_directory, 'versions')) as versions_file:
-            lines = versions_file.read().splitlines()
-            version_branches = list(filter(None, lines))
-
-        for branch in version_branches:
-            branch_output = path.join(output_base, branch)
-            branch_dir = tempfile.mkdtemp()
-            Repo.clone_from(base_directory, branch_dir, branch=branch)
-            branch_paths.append(
-                (branch_dir, branch_output)
-            )
-    else:
-        branch_paths.append(
-            (base_directory, output_base)
-        )
+    # Use tmp directory in home folder
+    home_temp = path.join(path.expanduser('~'), 'tmp')
```

Comment:
@WillMoggridge is there a built-in python library for using that cache directory?

I'm not using `/tmp` because the snap only has access to the user's home directory.